src/ctia/entity/investigation.clj

```diff
@@ -1,60 +1,38 @@
-             [crud :refer [entity-crud-routes]]]
-            [ctia.schemas
-             [utils :as csu]
-             [core :refer [def-stored-schema
-                           CTIAEntity]]
-             [sorting :as sorting]]
-            [ctia.schemas.graphql
-             [sorting :as graphql-sorting]
-             [flanders :as flanders]
-             [helpers :as g]
-             [pagination :as pagination]]
-            [ctia.stores.es
-             [mapping :as em]
-             [store :refer [def-es-store]]]
-            [ctim.schemas.investigation :as inv]
-            [flanders
-             [schema :as f-schema]
-             [spec :as f-spec]
-             [utils :as fu]]
-            [schema-tools.core :as st]
-            [schema.core :as s]
-            [ctia.schemas.graphql.ownership :as go]))
-
-(s/defschema Investigation
-  (st/merge (f-schema/->schema inv/Investigation)
-            CTIAEntity
-            {s/Keyword s/Any}))
-
-(f-spec/->spec inv/Investigation "investigation")
-
-(s/defschema PartialInvestigation
-  (st/merge (f-schema/->schema (fu/optionalize-all inv/Investigation))
-            CTIAEntity
-            {s/Keyword s/Any}))
-
-(s/defschema PartialInvestigationList
-  [PartialInvestigation])
-
-(s/defschema NewInvestigation
-  (st/merge (f-schema/->schema inv/NewInvestigation)
-            CTIAEntity
-            {s/Keyword s/Any}))
-
-(f-spec/->spec inv/NewInvestigation "new-investigation")
-
-(def-stored-schema StoredInvestigation Investigation)
-
-(s/defschema PartialStoredInvestigation
-  (csu/optional-keys-schema StoredInvestigation))
-
-(def realize-investigation
-  (default-realize-fn "investigation" NewInvestigation StoredInvestigation))
+  (:require
+   [ctia.http.routes
+    [common :refer [BaseEntityFilterParams
+                    PagingParams
+                    SourcableEntityFilterParams]]
+    [crud :refer [entity-crud-routes]]]
+   [ctia.stores.es
+    [mapping :as em]
+    [store :refer [def-es-store]]]
+   [ctia.entity.investigation.schemas
+    :refer [Investigation
+            PartialInvestigation
+            PartialStoredInvestigation
+            NewInvestigation
+            StoredInvestigation
+            PartialInvestigationList
+            realize-investigation]]
+   [schema-tools.core :as st]
+   [schema.core :as s]
+   [ctia.schemas.sorting :as sorting]))
+
+(def snapshot-action-fields-mapping
+  {:object_ids {:type "text"
+                :analyzer "text_analyzer"
+                :search_quote_analyzer "text_analyzer"
+                :search_analyzer "search_analyzer"
+                :include_in_all false}
+   :targets {:type "nested"
+             :include_in_all false}
+   :investigated_observables {:type "text"
+                              :analyzer "text_analyzer"
+                              :search_quote_analyzer "text_analyzer"
+                              :search_analyzer "search_analyzer"
+                              :include_in_all false}})
```

Comment:
We have specific mapping for observable, you should use this one:
https://github.com/threatgrid/ctia/blob/master/src/ctia/stores/es/mapping.clj#L130

Reply:
This field is not for full observables; its for a sequence of strings of the form `"type:value"`. Updating the examples will make that clear. The reason why its a sequence of strings has been discussed [here](https://github.com/threatgrid/ctim/pull/282#pullrequestreview-369379556).

Reply:
Ok, I remember, thus you should keep it as a text.
Please, directly use [this variable ](https://github.com/threatgrid/ctia/blob/master/src/ctia/stores/es/mapping.clj#L33) in for all your text mapping (there is other to replace in this PR).
It is the same value, but, in addition of avoiding code duplication, if we need to change its behavior it will be propagated to all text values.

Reply:
Ok. Done. I swapped in `em/text` and `em/token`.